src/address/normalize-address.ts

```diff
@@ -0,0 +1,65 @@
+import { EventEmitter } from "../isomorphic.node";
+import { NormalizedConfig } from "../config";
+import {
+  AddressValidateParams,
+  AddressValidateResult,
+  callJsonRpcMethod,
+} from "../json-rpc";
+import { formatAddress } from "./format-address";
+import {
+  Address,
+  AddressValidationResult,
+  NormalizedAddress,
+} from "./public-types";
+import { validateAddress } from "./validate-address";
+import { validateInputAddress } from "./validate-input-address";
+import { ShipEngineError } from "../errors";
+import { ErrorCode, ErrorType } from "../enums";
+
+/**
+ * Normalizes an address and returns it.
+ */
+export async function normalizeAddress(
+  address: Address,
+  config: NormalizedConfig,
+  events: EventEmitter
+  // @ts-expect-error TypeScript is confused by the return in the for loop
+): Promise<NormalizedAddress | ShipEngineError> {
```

Comment:
The function should only return a `Promise<NormalizedAddress>`.  Errors should be thrown, not returned.

Reply:
I was confused by the ticket and should have asked. Also, I was reading about how to properly do this in TS and found this blog post that talked about throwing vs returning errors. 

https://medium.com/inato/expressive-error-handling-in-typescript-and-benefits-for-domain-driven-design-70726e061c86

I will refactor it to throw though.